### ISSUE TITLE
Membership check regression in 0.0.61 -> 0.0.62

### ISSUE THREAD

**Issue by Pluviolithic** [astral-sh/ty#4066]
### Summary

For example:

```py
from typing import reveal_type

strings: set[str] = set()


def func(value: str | None) -> None:
    if value in strings:
        reveal_type(value) # -> Revealed type: `str | None`
        strings.remove(value) # -> Argument to bound method `set.remove` is incorrect...
```

[Playground](https://play.ty.dev/157f5609-75f0-4d77-a527-09e6575ac46f)

Full diagnostic output:

```
❯ ty check
info[revealed-type]: Revealed type
 --> example.py:8:21
  |
8 |         reveal_type(value)
  |                     ^^^^^ `str | None`
  |

error[invalid-argument-type]: Argument to bound method `set.remove` is incorrect
 --> example.py:9:24
  |
9 |         strings.remove(value)
  |                        ^^^^^ Expected `str`, found `str | None`
  |
info: element `None` of union `str | None` is not assignable to `str`
info: Method defined here
    --> stdlib/builtins.pyi:3352:9
     |
3352 |     def remove(self, element: _T, /) -> None:
     |         ^^^^^^       ----------- Parameter declared here
     |

Found 2 diagnostics
```

This does not occur with `uvx ty@0.0.61 check`.

### Version

ty 0.0.62

**Comment by AlexWaygood** [comment]
Thank you. Please see #3419
